cpp/src/io/utilities/file_io_utilities.cpp

```diff
@@ -75,8 +75,8 @@ class cufile_shim {
 
   ~cufile_shim()
   {
-    driver_close();
-    dlclose(cf_lib);
+    if (driver_close) driver_close();
+    if (cf_lib) dlclose(cf_lib);
```

Comment:
I approved #9784 for this fix. https://github.com/rapidsai/cudf/pull/9784#pullrequestreview-817858016